### PR TITLE
feat: increment route update for radixtree host uri, radixtree uri and radi…

### PR DESCRIPTION
**Pull request by ranxuxin001**
This PR extends apisix to modify a route without create the whole radixtree everytime.

refer to #9334

### CODE REVIEW

File: apisix/http/router/radixtree_uri.lua [hunk 1]
```diff
@@ -18,24 +18,128 @@ local require = require
...
 
-    local uri_routes = {}
-    local uri_router
-    local match_opts = {}
+local function incremental_operate_radixtree(routes)
+    local sync_tb = apisix_router.sync_tb
+    if apisix_router.need_create_radixtree then
+        uri_router = base_router.create_radixtree_uri_router(routes, uri_routes, false)
+        apisix_router.need_create_radixtree = false
+        for k, _ in pairs(sync_tb) do
```
Comment (kingluo) [inline apisix/http/router/radixtree_uri.lua:39]: Use require("table.clear") instead?

**Comment by Revolyssup** [comment]
ping @ranxuxin001

**Comment by nitishfy** [comment]
@ranxuxin001 are you available to work on this? We understand that you will be having other things to work for. In case you're not available to work on this, please mention it so that someone else can take this issue up!

Thank you again for contributing to apisix!

**Comment by wklken** [comment]
Perhaps this pull request should be reconsidered for reopening. I believe it could be highly beneficial for environments where the route configuration frequently changes.

**Comment by sergey-jr** [comment]
@Revolyssup Could you please reopen this issue? The problem still persists and seems serious. While there is a potential workaround with this patch: https://github.com/apache/apisix/issues/12466, an official fix from the project would be much appreciated.

**Comment by Revolyssup** [comment]
> @Revolyssup Could you please reopen this issue? The problem still persists and seems serious. While there is a potential workaround with this patch: #12466, an official fix from the project would be much appreciated.

@Baoyuantop Please take a look

**Comment by Baoyuantop** [comment]
Hi @sergey-jr, thanks for your feedback, I reopened it.

**Comment by Baoyuantop** [comment]
Since the original author is no longer able to work on this PR, anyone is welcome to resubmit.

**Comment by Baoyuantop** [comment]
Hi @ranxuxin001, following up on the previous review comments. Let us know if you have an update. Thanks!

**Comment by apaloleg** [comment]
@Baoyuantop Hi! I open new MR #12826 for this issue

**Comment by Baoyuantop** [comment]
Due to a lack of activity, this PR is closed for now. We will continue working on this issue in #12826. Thank you for your contribution @ranxuxin001 .

**Pull request closed without merging**